libraries/plugins/witness/witness.cpp

```diff
@@ -83,35 +88,62 @@ std::string witness_plugin::plugin_name()const
    return "witness";
 }
 
+void witness_plugin::add_private_key(const std::string& key_id_to_wif_pair_string)
+{
+   auto key_id_to_wif_pair = graphene::app::dejsonify<std::pair<chain::public_key_type, std::string> >(key_id_to_wif_pair_string, 5);
+   fc::optional<fc::ecc::private_key> private_key = graphene::utilities::wif_to_key(key_id_to_wif_pair.second);
+   if (!private_key)
+   {
+      // the key isn't in WIF format; see if they are still passing the old native private key format.  This is
+      // just here to ease the transition, can be removed soon
+      try
+      {
+         private_key = fc::variant(key_id_to_wif_pair.second, 2).as<fc::ecc::private_key>(1);
+      }
+      catch (const fc::exception&)
+      {
+         FC_THROW("Invalid WIF-format private key ${key_string}", ("key_string", key_id_to_wif_pair.second));
+      }
+   }
+
+   if (_private_keys.find(key_id_to_wif_pair.first) == _private_keys.end())
+   {
+      ilog("Public Key: ${public}", ("public", key_id_to_wif_pair.first));
+      _private_keys[key_id_to_wif_pair.first] = *private_key;
+   }
+}
+
 void witness_plugin::plugin_initialize(const boost::program_options::variables_map& options)
 { try {
    ilog("witness plugin:  plugin_initialize() begin");
    _options = &options;
    LOAD_VALUE_SET(options, "witness-id", _witnesses, chain::witness_id_type)
 
-   if( options.count("private-key") )
+   if (options.count("private-key"))
```

Comment:
Why this change? :P

Reply:
Well, I consider the `if(` style to be really bad. It was close to other changes... Reverted ;).